src/psbt.h

```diff
@@ -580,4 +610,12 @@ bool FinalizeAndExtractPSBT(PartiallySignedTransaction& psbtx, CMutableTransacti
  */
 NODISCARD TransactionError CombinePSBTs(PartiallySignedTransaction& out, const std::vector<PartiallySignedTransaction>& psbtxs);
 
+/**
+ * Provides helpful miscellaneous information about where a PSBT is in the signing workflow.
+ *
+ * @param[in] psbtx the PSBT to analyze
+ * @return A PSBTAnalysis with information about the provided PSBT.
+ */
+PSBTAnalysis AnalyzePSBT(PartiallySignedTransaction psbtx);
```

Comment:
`const PartiallySignedTransaction& psbtx`?

Reply:
I tried, but I had to back it out, because of the pattern of using "SignPSBTInput(DUMMY_SIGNING_PROVIDER, ...)" to get signature data without modifying the PSBT. So we either have to take the psbt by value, or by non-const reference.

We could get around this with a const cast, but I'm against it unless we think the cost of copying a PartiallySignedTransaction is unacceptable.

Reply:
Makes sense to keep by value. The code could be refactored but at the end each input would have to be copied to be able to call `FillSignatureData()` and `FromSignatureData`.